### PR TITLE
ROCm 5.1.3 scilibs, using rocm.eclass

### PR DESCRIPTION
**Pull request by littlewu2508**

### CODE REVIEW

File: dev-util/Tensile/Tensile-5.1.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+SLOT="0/$(ver_cut 1-2)"
+
+# Not compatible with recent versions of pytest
+RESTRICT="test"
+
+RDEPEND="${PYTHON_DEPS}
+	dev-python/pyyaml[${PYTHON_USEDEP}]
+	dev-python/msgpack[${PYTHON_USEDEP}]
+	>=dev-util/rocm-smi-4.3.0
+	>=sys-devel/clang-14.0.5-r1:${LLVM_MAX_SLOT}=
```
Comment (heroxbd) [inline dev-util/Tensile/Tensile-5.1.3.ebuild:29]: This is strange to say the version should larger than some `MAX`.

File: sci-libs/rocBLAS/rocBLAS-5.1.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+REQUIRED_USE="${ROCM_REQUIRED_USE}"
+RESTRICT="!test? ( test )"
+
+BDEPEND="
+	dev-util/rocm-cmake:${SLOT}
+	dev-util/Tensile:${SLOT}
+"
+
+DEPEND="
+	dev-util/hip:${SLOT}
```
Comment (heroxbd) [inline sci-libs/rocBLAS/rocBLAS-5.1.3.ebuild:31]: Compared to 5.0.2, why do you lock the hip version?

```patch
 DEPEND="                  
-       dev-util/hip:=               
+       dev-util/hip:${SLOT}                                                                              
        dev-libs/msgpack                                                                                                                                                                                            
        test? (                                    
                virtual/blas                      
```

File: sci-libs/rocFFT/rocFFT-5.1.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	dev-texlive/texlive-latex
+	dev-tex/latexmk
+	sys-apps/texinfo
+	dev-python/sympy[${PYTHON_USEDEP}]
+	dev-python/numpy[${PYTHON_USEDEP}]
+	dev-python/scipy[${PYTHON_USEDEP}]
+	dev-python/pandas[${PYTHON_USEDEP}] )
+${PYTHON_DEPS}"
+
+DEPEND="dev-util/hip:${SLOT}
```
Comment (heroxbd) [inline sci-libs/rocFFT/rocFFT-5.1.3.ebuild:32]: Same as rocBLAS.

File: sci-libs/rocThrust/rocThrust-5.1.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+LICENSE="Apache-2.0"
+KEYWORDS="~amd64"
+SLOT="0/$(ver_cut 1-2)"
+IUSE="benchmark test"
+REQUIRED_USE="${ROCM_REQUIRED_USE}"
+
+RESTRICT="!test? ( test )"
+
+RDEPEND="dev-util/hip
```
Comment (heroxbd) [inline sci-libs/rocThrust/rocThrust-5.1.3.ebuild:22]:
```patch
@@ -13,13 +15,14 @@                                                                                      
 KEYWORDS="~amd64"       
 SLOT="0/$(ver_cut 1-2)"
 IUSE="benchmark test"
+REQUIRED_USE="${ROCM_REQUIRED_USE}"
                                                    
 RESTRICT="!test? ( test )"                                                                              
                                                    
-RDEPEND="dev-util/hip:=                                                                                 
-       sci-libs/rocPRIM:${SLOT}"                                                                        
-DEPEND="${RDEPEND}
+RDEPEND="dev-util/hip
+       sci-libs/rocPRIM:${SLOT}[${ROCM_USEDEP}]
        test? ( dev-cpp/gtest )"     
+DEPEND="${RDEPEND}"
 BDEPEND=">=dev-util/cmake-3.22"
             
 S="${WORKDIR}/rocThrust-rocm-${PV}"
```

Put in your commit message why there is no need to rebuild rocThrust if hip is upgraded.

File: dev-libs/rccl/rccl-5.1.3.ebuild [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+DESCRIPTION="ROCm Communication Collectives Library (RCCL)"
+HOMEPAGE="https://github.com/ROCmSoftwarePlatform/rccl"
+SRC_URI="https://github.com/ROCmSoftwarePlatform/rccl/archive/rocm-${PV}.tar.gz -> rccl-${PV}.tar.gz"
+
+LICENSE="BSD"
+KEYWORDS="~amd64"
+SLOT="0/$(ver_cut 1-2)"
+IUSE="test"
+
+RDEPEND="dev-util/hip
```
Comment (heroxbd) [inline dev-libs/rccl/rccl-5.1.3.ebuild:19]: Explain in your commit message that there is no need to lock hip version.

```
-RDEPEND="dev-util/hip:${SLOT}"                                                                           
+RDEPEND="dev-util/hip         
+dev-util/rocm-smi:${SLOT}"
 DEPEND="${RDEPEND}"    
 BDEPEND=">=dev-util/cmake-3.22                                                                           
 ```

File: dev-util/Tensile/Tensile-5.1.3.ebuild (hunk 1 above)
Comment (littlewu2508) [reply dev-util/Tensile/Tensile-5.1.3.ebuild:29]: No, it's not a version larger than `LLVM_MAX_SLOT`. `LLVM_MAX_SLOT` acts here as a slot, which means Tensile depend on this slot of clang. The `>=` is to specify the min clang patch version.

File: sci-libs/rocBLAS/rocBLAS-5.1.3.ebuild (hunk 2 above)
Comment (littlewu2508) [reply sci-libs/rocBLAS/rocBLAS-5.1.3.ebuild:31]: I made a mistake here. Now rocBLAS is free of hip version.

File: dev-libs/rccl/rccl-5.1.3.ebuild (hunk 5 above)
Comment (littlewu2508) [reply dev-libs/rccl/rccl-5.1.3.ebuild:19]: Done

File: sci-libs/rocThrust/rocThrust-5.1.3.ebuild (hunk 4 above)
Comment (littlewu2508) [reply sci-libs/rocThrust/rocThrust-5.1.3.ebuild:22]: Done

Comment (littlewu2508) [reply sci-libs/rocThrust/rocThrust-5.1.3.ebuild:22]: Also added similar information in rocBLAS/rocFFT commit

File: dev-util/Tensile/Tensile-5.1.3.ebuild (hunk 1 above)
Comment (heroxbd) [reply dev-util/Tensile/Tensile-5.1.3.ebuild]: OK, looks good.

**Pull request closed without merging**